Sources/SwiftProtobuf/Message+StreamDecodingIterator.swift

```diff
@@ -0,0 +1,131 @@
+// Sources/SwiftProtobuf/Message+StreamDecodingIterator.swift - Iterator over binary delimited input streams
+//
+// Copyright (c) 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Extensions to `Message` to provide binary coding and decoding using ``Foundation/Data``.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+extension Message {
+  
+  public static func streamDecodingIterator(inputStream: InputStream, bufferLength: Int = 32768) -> StreamDecodingIterator<Self> {
```

Comment:
Maybe do a file scoped constant for the two public apis that default the buffer size?

Reply:
Done